### PR TITLE
Fix retry logic while updating cluster status in e2e test

### PR DESCRIPTION
**Pull request by shashidharatd**
Fixes: https://github.com/kubernetes-sigs/federation-v2/issues/839#issuecomment-492078999

/cc @marun @irfanurrehman

### CODE REVIEW

File: test/e2e/servicedns.go [hunk 1]
```diff
@@ -219,7 +217,11 @@ func createClusterServiceAndEndpoints(f framework.FederationFramework, name, nam
...
 
-func ensureClustersHaveRegionZoneAttributes(tl common.TestLogger, client genericclient.Client, federatedClusters *fedv1a1.FederatedClusterList) map[string]fedv1a1.FederatedClusterStatus {
+func ensureClustersHaveRegionZoneAttributes(tl common.TestLogger, client genericclient.Client, federationSystemNamespace string) map[string]fedv1a1.FederatedClusterStatus {
+	federatedClusters := &fedv1a1.FederatedClusterList{}
+	err := client.List(context.TODO(), federatedClusters, federationSystemNamespace)
+	framework.ExpectNoError(err, "Error listing federated clusters")
+
 	clusterRegionZones := make(map[string]fedv1a1.FederatedClusterStatus)
 	for i, cluster := range federatedClusters.Items {
 		cluster.Status.Region = fmt.Sprintf("r%d", i)
```
Comment (marun) [inline test/e2e/servicedns.go]: The status needs to be set inside the wait loop to ensure that the resource retrieved on conflict is updated.

Comment (shashidharatd) [reply test/e2e/servicedns.go]: @marun, this location is for fetching cluster list. Below i am fetching the individual cluster if there is a conflict.

Comment (shashidharatd) [reply test/e2e/servicedns.go]: sorry, got your comment.

Comment (shashidharatd) [reply test/e2e/servicedns.go]: Fixed now, PTAL.

**Pull request merged**